Samples/excel-add-in-shape-dashboard/src/taskpane/taskpane.ts

```diff
@@ -0,0 +1,431 @@
+
+    let category = shapeGroup.items[3];
+    category.textFrame.textRange.getSubstring(0, 1).font.size = 30;
+    category.textFrame.textRange.getSubstring(1).font.size = 12;
+    category.textFrame.textRange.getSubstring(1).font.color = "#99CCFF";
+
+    let subCategory = shapeGroup.items[4];
+    subCategory.textFrame.textRange.getSubstring(0, 2).font.size = 30;
+    subCategory.textFrame.textRange.getSubstring(2).font.size = 12;
+    subCategory.textFrame.textRange.getSubstring(2).font.color = "#99CCFF";
+
+    let maxSale = shapeGroup.items[6];
+    maxSale.textFrame.textRange.getSubstring(0, 8).font.size = 18;
+    maxSale.textFrame.textRange.getSubstring(8).font.size = 13;
+    maxSale.textFrame.textRange.getSubstring(8).font.color = "FFFF66";
+
+    let sumSale = shapeGroup.items[7];
+    sumSale.textFrame.textRange.getSubstring(0, 9).font.size = 15;
+    sumSale.textFrame.textRange.getSubstring(9).font.size = 13;
+    sumSale.textFrame.textRange.getSubstring(9).font.color = "FFFF66";
+
+    await context.sync();
+    showStatus("Change information format - success!", false);
+  });
+}
+
+async function onActivate() {
+  await Excel.run(async (context) => {
+    let shapes = context.workbook.worksheets.getItem("Sample").shapes;
+    shapes.load("items");
+    await context.sync();
+    let shapeGroup = shapes.items[0].group.shapes;
+    shapeGroup.load("items");
+    await context.sync();
+
+    let region = shapeGroup.items[0];
+    region.textFrame.textRange.text = "-Central\n-South\n-West\n-East";
+    region.textFrame.horizontalAlignment = Excel.HorizontalAlignment.left;
+    region.textFrame.textRange.font.size = 13;
+
+    let state = shapeGroup.items[1];
+    state.textFrame.textRange.text = "Top 3:\n-California\n-Washington\n-New York";
+    state.textFrame.horizontalAlignment = Excel.HorizontalAlignment.left;
+    state.textFrame.textRange.font.size = 12;
+
+    let category = shapeGroup.items[3];
+    category.textFrame.textRange.text = "-OfficeSupply\n-Furniture\n-Technology";
+    category.textFrame.horizontalAlignment = Excel.HorizontalAlignment.left;
+    category.textFrame.textRange.font.size = 10;
+
+    let subCategory = shapeGroup.items[4];
+    subCategory.textFrame.textRange.text = "Top 3:\n-Phone\n-Tables\n-Accessories";
+    subCategory.textFrame.horizontalAlignment = Excel.HorizontalAlignment.left;
+    subCategory.textFrame.textRange.font.size = 10;
+
+    let maxScale = shapeGroup.items[6];
+    maxScale.textFrame.textRange.text = "US2015126214\n-Furniture\n-Washington";
+    maxScale.textFrame.textRange.font.size = 11;
+
+    let subScale = shapeGroup.items[7];
+    subScale.textFrame.textRange.text = "Top 3:\nUS2015126214\nCA2017137596\nCA2014115812";
+    subScale.textFrame.horizontalAlignment = Excel.HorizontalAlignment.left;
+    subScale.textFrame.textRange.font.size = 11;
+
+    await context.sync();
+  });
+}
+
+async function onDeactivate() {
+  await changeColor();
+  await changeFontFormat();
```

Comment:
Doesn't this cause the event listeners to be added every time the event triggers? That seems like a performance issue.